### PR TITLE
Add support for .net global tool configs

### PR DESCRIPTION
**Pull request by xt0rted**
### Checklist:

- [x] If this PR is a new feature, please provide at least one example link
- [x] Make sure all of the significant new logic is covered by tests

**Sample files**

- https://github.com/dotnet/runtime/blob/ac68cdeeea2abbff988448fe7915f2bcf5ee3cfd/.config/dotnet-tools.json
- https://github.com/dotnet/roslyn/blob/e3a461796d3cb4917367af73778d2ab34f6812e9/dotnet-tools.json

The changes to the regex builder are needed to allow turning off the `g` flag because some tools have commands of the same name, such as in the second link, and both instances of `dotnet-format` were being linkified.

### CODE REVIEW

File: packages/helper-regex-builder/index.js [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
+  return createRegEx(`${keyField}\\s*:\\s*${valueField}`, globalSearch);
 }
 
 export function jsonRegExKeyValue(key, value) {
-  return regexBuilder(key, value, true);
+  return regexBuilder(key, value, true, true);
+}
+
+export function jsonRegExSingleKeyValue(key, value) {
+  return regexBuilder(key, value, true, false);
```
Comment (stefanbuck) [inline packages/helper-regex-builder/index.js]: Personally I would prefer adding another argument to `jsonRegExKeyValue` instead of introducing another function. Basically how it was until https://github.com/OctoLinker/OctoLinker/commit/93d85161cd3eacac9ad467dcdd55075a2566a224#diff-b0829a6c903b7a1200ba3d8cfbf4f331

Comment (xt0rted) [reply packages/helper-regex-builder/index.js]: That's actually the exact setup I had but I switched to this thinking it'd be easier to maintain. Now it makes sense as to why  there's unit tests with a 3rd parameter for those functions even though they only have 2.

Comment (stefanbuck) [reply packages/helper-regex-builder/index.js]: Up to you, I see value in both approaches ... at the end it doesn't really matter 😉 

Regarding the tests, in #774 I had to change quite a lot so it sneaked through ... sorry for the confusion

**Comment by xt0rted** [comment]
While testing this I noticed that it's turning the `//` item into a link to  nuget.org. I wasn't able to find why this was happening though since `//` clearly isn't a valid package on the site.

https://github.com/xt0rted/OctoLinker/blob/c643e7de5dcfc0b0665a54af361fe8c44f331c8c/e2e/fixtures/dotnet/dotnet-tools.json#L5

**Comment by stefanbuck** [comment]
The RegExp builder returns the follow RegExp

```
("//")\s*:\s*"(@OctoLinkerResolve\(https://www\.nuget\.org/packages/dotnet\-format\))"
```

After applying the RegExp the resolver gets called  with `["//", "@OctoLinkerResolve(https://www.nuget.org/packages/dotnet-format)"]` where the first element will become the `target` variable. The url of this element is `https://www.nuget.org/packages///`.

We could add a conditional to prevent this but maybe not worth adding unnecessary complexity since it's not a real issue. What do you think?

**Comment by xt0rted** [comment]
I can live with it if it's just in the test files. No need to over complicate things for an edge case.
